app/src/main/java/seemoo/fitbit/activities/MainActivity.java

```diff
@@ -34,13 +38,14 @@
 /**
  * The main menu.
  */
-public class MainActivity extends AppCompatActivity {
+public class MainActivity extends RequestPermissionsActivity {
 
     private final String TAG = this.getClass().getSimpleName();
 
     private static final int REQUEST_ENABLE_BLUETOOTH = 1;
     private static final int REQUEST_ACCESS_FINE_LOCATION = 1;
-    private static final int REQUEST_EXTERNAL_STORAGE = 1;
+    private static final int REQUEST_EXTERNAL_STORAGE = 2;
+    private static final int REQUEST_APP_SETTINGS = 1;

```

Comment:
both could just use a protected variant from RequestPermissionsActivity, or am I wrong?

Reply:
Should work, going to test and adjust today.